### DISCUSSION TITLE
Allow returning custom error bodies

### DISCUSSION THREAD

**Discussion by Cogitri** [maxcountryman/tower-sessions#204] [Ideas]
- [X] I have looked for existing issues (including closed) about this

## Feature Request

It'd be nice of tower-sessions had support for returning custom error HTTP bodies. We use tower-sessions in a microservice and want to return more detailed error informations to our main service in case of errors - currently tower-session just logs the error and then sends an HTTP 500 with an empty body when something goes wrong in the ServiceStore (e.g. no DB connection).

### Version

0.12.2

**Comment by maxcountryman** [comment]
I’d be happy to review PRs that implement this. I’d suggest using `tower-http` as a model for custom responses (they use a callback which registers a handler for responses, as I recall).

**Reply by Cogitri** [reply]
Alright, thanks for the feedback. I've added that to our backlog :)

**Reply by Cogitri** [reply]
Hej, just looking at this again, could you specify which callback you're referring to? They seem to offer quite a lot of callbacks and I figured asking is best before implementing something that doesn't fit for tower-sessions :)

**Reply by maxcountryman** [reply]
I think using the `trace` service as a rough model is probably where to start: https://docs.rs/tower-http/latest/tower_http/trace/struct.DefaultOnFailure.html

In that example, there's an `on_failure` method that's used to register the callback.

I'm not sure if this would need to be adapted for our specific use case but hopefully that illustrates the idea.
